src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsremandandsentencingapi/jpa/entity/PeriodLengthEntity.kt

```diff
@@ -59,7 +59,8 @@ class PeriodLengthEntity(
     days == other?.days &&
     periodOrder == other?.periodOrder &&
     periodLengthType == other.periodLengthType &&
-    legacyData == other.legacyData
+    legacyData == other.legacyData &&
+    statusId == other.statusId
```

Comment:
I added status-id as well to the isSame check.. @carloveo-moj can you double check this as the isSame method is also used by the non-legacy upserts

reason for adding is that the statusId can change atm from the update legacy route